Userland/Libraries/LibWeb/Infra/ByteSequences.cpp

```diff
@@ -25,4 +25,35 @@ void byte_uppercase(ByteBuffer& bytes)
         bytes[i] = to_ascii_uppercase(bytes[i]);
 }
 
+// https://infra.spec.whatwg.org/#byte-sequence-prefix
+// https://infra.spec.whatwg.org/#byte-sequence-starts-with
+bool is_prefix_of(ByteBuffer const& potential_prefix, ByteBuffer const& input)
```

Comment:
it seems that itʼd be good for the use case to return `the smallest index such that the nth byte of a is different from the nth byte of b` you use in step 3. of `is_byte_less_than` – but thatʼs not a big deal (ignore this if not really interested).